crypto/src/hash/poseidon/mod.rs

```diff
@@ -70,6 +70,7 @@ impl<P: PermutationParameters> Poseidon for P {
         x.clone()
     }
 
+    #[allow(clippy::manual_div_ceil)]
```

Comment:
Why not just follow the advice? It seems it's complaining about line 80:
```rust
values.resize(((values.len() + r - 1) / r) * r, FE::zero());
```
This could be:
```rust
values.resize(values.len().div_ceil(r), FE::zero());
```

Reply:
Also, the `assert` below should probably go (the `_ceil` part implies it) or at least become `debug_assert` instead.

Reply:
Actually, it's:
```rust
values.resize(values.len().div_ceil(r) * r, FE::zero());
```

Reply:
Yes, we fixed it [here](https://github.com/lambdaclass/lambdaworks/pull/944/commits/cf35df8e506cd61bf54d872de4002b99f64ce9a3).